tests/test_datasets/test_dataset.py

```diff
@@ -121,17 +140,16 @@ def test_dataset_check_exog_raise_error(exog_starts_later: bool, exog_ends_earli
         TSDataset._check_exog(df=df, df_exog=dfexog)
 
 
-def test_dataset_check_exog_pass():
-    timestamp = pd.date_range("2021-01-01", "2021-02-01")
-    df1 = pd.DataFrame({"timestamp": timestamp, "target": 11, "segment": "1"})
-    df2 = pd.DataFrame({"timestamp": timestamp[5:], "target": 12, "segment": "2"})
-    df = pd.concat([df1, df2], ignore_index=True)
-    df = TSDataset.to_dataset(df)
+def test_dataset_check_exog_pass(df_and_regressors):
+    df, df_exog = df_and_regressors
+    _ = TSDataset._check_exog(df=df, df_exog=df_exog)
 
-    timestamp = pd.date_range("2021-01-01", "2021-02-11")
-    df1 = pd.DataFrame({"timestamp": timestamp, "regressor_aaa": 1, "segment": "1"})
-    df2 = pd.DataFrame({"timestamp": timestamp[5:], "regressor_aaa": 2, "segment": "2"})
-    dfexog = pd.concat([df1, df2], ignore_index=True)
-    dfexog = TSDataset.to_dataset(dfexog)
 
-    _ = TSDataset._check_exog(df=df, df_exog=dfexog)
+def test_warn_not_enough_exog(df_and_regressors, recwarn):
+    """Check that warning is thrown if regressors don't have enough values."""
+    df, df_exog = df_and_regressors
+    ts = TSDataset(df=df, df_exog=df_exog, freq="D")
+    ts.make_future(ts.df_exog.shape[0] + 100)
+    assert len(recwarn) == 2
+    for warn in recwarn:
+        assert str(warn.message).startswith("Some regressors don't have enough values")
```

Comment:
Is it possible to use https://docs.pytest.org/en/6.2.x/warnings.html#warns here?

Reply:
Added in new commit.